### PR TITLE
fix: altered tf.config function call to be compatible with TF 1.15 [DET-4852]

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
Running distributed training on TF 1.15 in eager execution mode resulted in an AttributeError as the attribute currently being used is only available in TF 2.0 and higher. This changes the function call such that it is also compatible with version 1.15.

## Test Plan
Manually tested

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by eecsliu** [comment]
`tf.config.experimental.set_visible_devices` exists in tensorflow 2, but should I still do a version check and only have tensorflow 1.15 use this experimental line?

**Review by rb-determined-ai — COMMENTED** [review]
Does this work in the latest tensorflow as well?

If so... lol, I had no idea we were so close to eager execution on keras 1.15

**Comment by eecsliu** [comment]
> Does this work in the latest tensorflow as well? If so, I don't see a reason to have extra version checks.
> 
> If so... lol, I had no idea we were so close to eager execution on keras 1.15

.....was eager execution broken before? I tested it on the tf-1.15 and tf-2.2 environments and it worked for both. For 1.15 I had to set it to eager mode and add the argument `experimental_run_tf_function=False` in model.compile, but it worked after that.

**Review by rb-determined-ai — APPROVED** [review]
I verified that this API is available at least until TF 2.4
